server/locks_controller_test.go

```diff
@@ -192,6 +194,46 @@ func TestDeleteLock_OldFormat(t *testing.T) {
 	cp.VerifyWasCalled(Never()).CreateComment(AnyRepo(), AnyInt(), AnyString())
 }
 
+func TestDeleteLock_UpdateProjectStatus(t *testing.T) {
+	t.Log("When deleting a lock, pull status has to be updated to reflect discarded plan")
+	RegisterMockTestingT(t)
+
+	repoName := "owner/repo"
+	projectPath := "path"
+	workspaceName := "workspace"
+
+	cp := vcsmocks.NewMockClient()
+	l := mocks.NewMockLocker()
+	workingDir := mocks2.NewMockWorkingDir()
+	workingDirLocker := events.NewDefaultWorkingDirLocker()
+	pull := models.PullRequest{
+		BaseRepo: models.Repo{FullName: repoName},
+	}
+	When(l.Unlock("id")).ThenReturn(&models.ProjectLock{
+		Pull:      pull,
+		Workspace: workspaceName,
+		Project: models.Project{
+			Path:         projectPath,
+			RepoFullName: repoName,
+		},
+	}, nil)
+	db := dbmocks.NewMockBoltDB()
```

Comment:
Can you use a real boltdb instead?

```go
	tmp, cleanup := TempDir(t)
        defer os.Remove(tmp)
	db, err := db.New(tmp)
	Ok(t, err)
```

Reply:
Sorry I don't follow.
The value of this test is to verify that db.UpdateProjectStatus is actually called when someone uses the deleteLock endpoint.
Using mock allows me to verify this (see line 234) in a simple way.

If I use a real DB like you suggest, how can I do this?
Am I missing something?

Reply:
Commented above.